### ISSUE TITLE
Getting dimensions error while using dynamic=True

### ISSUE THREAD

**Issue by VM8198** [ultralytics/ultralytics#19166]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello @glenn-jocher ,

I have exported my custom trained yolov9t model to tensorrt to run it in Jetson Orin Nano Super using below code.

```
from ultralytics import YOLO

model = YOLO("yolo9t.pt")

model.export(format="engine", half=True, device="dla:0", dynamic=True) 
```

It works fine for 1 rtsp stream. But when I use multiple streams using list.streams it throws below error.
`
[02/10/2025-20:05:47] [TRT] [E] IExecutionContext::setInputShape: Error Code 3: API Usage Error (Parameter check failed, condition: engineDims.d[i] == dims.d[i]. Static dimension mismatch while setting input shape for images. Set dimensions are [2,3,640,640]. Expected dimensions are [1,3,-1,-1].)
`

Can you please guide me if I'm doing anything wrong?

Thanks.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @VM8198, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for detailed information on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions and troubleshooting techniques are covered there.

It seems you're encountering a 🐛 **Bug or Error Report** with TensorRT and dynamic dimensions while using multiple streams. To help us debug this effectively, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)? This may include additional details like how you’re initializing and managing the multiple streams or the specific input sizes involved.

In the meantime, ensure that you are using the latest version of Ultralytics software with compatible dependencies. You can update to the latest release using the following command:

```bash
pip install -U ultralytics
```

Additionally, verify that your environment meets the latest [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

For real-time debugging, you might find it helpful to execute your workflows in one of these verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a> See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) for details.

## Resources

- For tips on exporting and troubleshooting TensorRT models, consult our [Export Documentation](https://docs.ultralytics.com/modes/export/) for best practices.
- Consider joining our [Discord Community](https://discord.com/invite/ultralytics) 🎧 for live discussions or checking out [Discourse Forums](https://community.ultralytics.com/) and our [Subreddit](https://reddit.com/r/Ultralytics) for more in-depth knowledge sharing.

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If all CI tests are 🎉 green, the latest version of Ultralytics should be verified to work correctly across all tasks and modes.

This is an automated response 🤖, but rest assured, an Ultralytics engineer will assist you soon. Thank you for your patience and for being a part of the Ultralytics community! ✨

**Comment by Y-T-G** [comment]
`dynamic=True` requires specifying maximum batch size using `batch` argument during export. You haven't specified any, so it's using 1 as maximum.

**Comment by VM8198** [comment]
Okay, so what batch size do I set? Should  it be the same number included in list.streams?

**Comment by VM8198** [comment]
Thanks @Y-T-G ,

it works fine now.
